File: 1762-buildings-with-an-ocean-view/1762-buildings-with-an-ocean-view.py
class Solution:
    def findBuildings(self, heights: List[int]) -> List[int]:
        if not heights:
            return []
        
        max_h = heights[-1]
        res = collections.deque([len(heights) - 1])
        
        for i in range(len(heights) - 2, -1, -1):
            curr_h = heights[i]
            if curr_h > max_h:
                max_h = curr_h
                res.appendleft(i)
        return res
        

File: 1762-buildings-with-an-ocean-view/1762-buildings-with-an-ocean-view.js
/**
 * @param {number[]} heights
 * @return {number[]}
 */
var findBuildings = function(heights) {
    if (heights.length === 0) {
        return []
    }
    let maxH = -1;
    const res = [];
    
    for (let i = heights.length - 1; i >= 0; i--) {
        if (heights[i] > maxH) {
            maxH = heights[i];
            res.push(i);
        }
    }
    
    return res.reverse();
    
};